frontend/src/api/calls/skillTypeCalls.ts

```diff
@@ -42,3 +48,26 @@ export async function getSkillTypeFromSkill(skill: IUserSkill | IProjectSkill):
 export function getAllSkillTypesFromLinks(url: string): Promise<ISkillType[]> {
     return <Promise<ISkillType[]>>getAllEntitiesFromLinksUrl(url, skillTypeCollectionName);
 }
+
+export function getAllSkillTypesFromPage(url: string): Promise<ISkillType[]> {
+    return <Promise<ISkillType[]>>getAllEntitiesFromPage(url, skillTypeCollectionName);
+}
+
+export async function getSkillTypeByName(skillName: string): Promise<ISkillType> {
+    const skills = await getAllSkillTypesFromPage(
+        getQueryUrlFromParams(apiPaths.skillTypesByName, {
+            name: skillName,
+        })
+    );
+    return skills.length == 0 ? await createNewSkill(skillName) : skills[0];
+}
+
+export function getRandomColor() {
+    let color = Math.floor(Math.random() * 16777216).toString(16);
+    return "#000000".slice(0, -color.length) + color;
+}
+
+export async function createNewSkill(skillName: string): Promise<ISkillType> {
+    const skill = new SkillType(skillName, getRandomColor());
+    return (await axios.post(apiPaths.skillTypes, skill, AxiosConf)).data;
+}
```

Comment:
I think we should have a call about this. I feel like we have very different ideas on what a skill and a skilltype is.